src/runtime_src/core/tools/common/ReportAieMem.cpp

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "ReportAieMem.h"
+#include "core/common/info_aie.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#define fmt4(x) boost::format("%4s%-22s: " x "\n") % " "
+#define fmt8(x) boost::format("%8s%-22s: " x "\n") % " "
+#define fmt12(x) boost::format("%12s%-22s: " x "\n") % " "
+#define fmt16(x) boost::format("%16s%-22s: " x "\n") % " "
+
+boost::property_tree::ptree
+populate_aie_mem(const xrt_core::device * _pDevice, const std::string& desc)
+{
+  xrt::device device(_pDevice->get_device_id());
+  boost::property_tree::ptree pt_mem;
+  pt_mem.put("description", desc);
+  std::stringstream ss;
+  ss << device.get_info<xrt::info::device::aie_mem>();
+  boost::property_tree::read_json(ss, pt_mem);
+
+  return pt_mem;
+}
+
+void
+ReportAieMem::getPropertyTreeInternal(const xrt_core::device * _pDevice, 
+                                      boost::property_tree::ptree &_pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data, 
+  // Then update this method to do so.
+  getPropertyTree20202(_pDevice, _pt);
+}
+
+void 
+ReportAieMem::getPropertyTree20202(const xrt_core::device * _pDevice, 
+                                   boost::property_tree::ptree &_pt) const
+{
+  _pt.add_child("aie_mem_status", populate_aie_mem(_pDevice, "Aie_Mem_Status"));
+}
+
+void 
+ReportAieMem::writeReport(const xrt_core::device* /*_pDevice*/,
+                          const boost::property_tree::ptree& _pt, 
+                          const std::vector<std::string>& _elementsFilter,
+                          std::ostream & _output) const
+{
+  boost::property_tree::ptree empty_ptree;
+  std::vector<std::string> aieTileList;
+
+  _output << "AIE\n";
+
+  // Loop through all the parameters given under _elementsFilter i.e. -e option
+  for (auto it = _elementsFilter.begin(); it != _elementsFilter.end(); ++it) {
+    // Only show certain selected tiles from aiemem that are passed under tiles
+    // Ex. -r aiemem -e tiles 0,3,5
+    if (*it == "tiles") {
+      auto tile_list = std::next(it);
+      if (tile_list != _elementsFilter.end())
+        boost::split(aieTileList, *tile_list, boost::is_any_of(","));
```

Comment:
We use element filter to **remove** elements. Not to add them in. It would be better to skip the tiles that are found within the filter.

Reply:
@dbenusov-xilinx I checked other reports like DebugIpStatus, Aie, AieShim, they are using elements filter to show information that are under filter. Please check ReportDebugIpStatus.cpp file.